### ISSUE TITLE
Can't leverage callbacks

### ISSUE THREAD

**Issue by crazyoptimist** [typpo/quickchart-ruby#3]
Quickchart is amazing @typpo !
How can I leverage callbacks in ruby client? I couldn't figure it out in any way.
Please advice.
Thanks in advance!

**Comment by typpo** [comment]
Thanks @CrazyOptimist!

Right now callbacks in Ruby are a bit clunky.  You'll have to specify your chart config as a string, like in this example: https://github.com/typpo/quickchart-ruby/blob/master/examples/simple_example_with_function.rb

If you want to keep the hash-based approach, another option is to add a placeholder string for your function, convert the object to json, and then find and replace the value of the placeholder with your function.

**Comment by crazyoptimist** [comment]
Oh my,
Why did I miss that example? 🤯  Text configuration is perfect for me. 👍 
Thanks a lot!

**Comment by luctus** [comment]
Hello @typpo. Thanks a lot for this project, it's simply perfect.

I'm having the same issue here. I want to keep the hash-based approach so I did it like this:

```rb
config = {
      type: "line",
      data: {
        labels: chart.labels,
        datasets: [
          {
            label: chart.series[0][:name],
            data: chart.series[0][:data],
            borderColor: chart.series[0][:color],
            backgroundColor: chart.series[0][:fill],
            lineTension: 0.2,
            borderWidth: 2,
            borderDash: [2,2],
            radius: 0
          },
          {
            label: chart.series[1][:name],
            data: chart.series[1][:data],
            borderColor: chart.series[1][:color],
            backgroundColor: chart.series[1][:fill],
            lineTension: 0.2,
            borderWidth: 2,
            radius: 1
          }
        ]
      },
      options: {
        legend: {
          position: 'bottom'
        },
        scales: {
          xAxes: [{
            gridLines: {
              display: true
            },
            ticks: {
              callback: "--callback--"
            }
            
          }]
        }
      }
    }
    config_json = config.to_json.gsub('--callback--', "(val) => {return val + 'k';}")
    
    qc = QuickChart.new(config_json, width: 700, height: 428, device_pixel_ratio: 2.0)
    
    puts qc.get_url
```

And I'm getting something an image that displays a "Char error: (val) => {return val + 'k';} is not a function" message.

https://quickchart.io/chart?c=%7B%22type%22%3A%22line%22%2C%22data%22%3A%7B%22labels%22%3A%5B%2207%2FJan%22%2C%2208%2FJan%22%2C%2209%2FJan%22%2C%2210%2FJan%22%2C%2211%2FJan%22%2C%2212%2FJan%22%2C%2213%2FJan%22%2C%2214%2FJan%22%2C%2215%2FJan%22%2C%2216%2FJan%22%2C%2217%2FJan%22%2C%2218%2FJan%22%2C%2219%2FJan%22%2C%2220%2FJan%22%2C%2221%2FJan%22%2C%2222%2FJan%22%2C%2223%2FJan%22%2C%2224%2FJan%22%2C%2225%2FJan%22%2C%2226%2FJan%22%2C%2227%2FJan%22%2C%2228%2FJan%22%2C%2229%2FJan%22%2C%2230%2FJan%22%2C%2231%2FJan%22%2C%2201%2FFeb%22%2C%2202%2FFeb%22%2C%2203%2FFeb%22%2C%2204%2FFeb%22%2C%2205%2FFeb%22%2C%2206%2FFeb%22%2C%2207%2FFeb%22%2C%2208%2FFeb%22%2C%2209%2FFeb%22%2C%2210%2FFeb%22%2C%2211%2FFeb%22%2C%2212%2FFeb%22%2C%2213%2FFeb%22%2C%2214%2FFeb%22%2C%2215%2FFeb%22%2C%2216%2FFeb%22%2C%2217%2FFeb%22%2C%2218%2FFeb%22%2C%2219%2FFeb%22%2C%2220%2FFeb%22%2C%2221%2FFeb%22%2C%2222%2FFeb%22%2C%2223%2FFeb%22%2C%2224%2FFeb%22%2C%2225%2FFeb%22%2C%2226%2FFeb%22%2C%2227%2FFeb%22%2C%2228%2FFeb%22%2C%2201%2FMar%22%2C%2202%2FMar%22%2C%2203%2FMar%22%2C%2204%2FMar%22%2C%2205%2FMar%22%2C%2206%2FMar%22%2C%2207%2FMar%22%5D%2C%22datasets%22%3A%5B%7B%22label%22%3A%22Sessions+in+the+last+year%22%2C%22data%22%3A%5B285138%2C246902%2C186709%2C215908%2C304702%2C296100%2C297337%2C285973%2C251015%2C184504%2C215682%2C296620%2C309188%2C292284%2C272930%2C260625%2C202952%2C219494%2C307932%2C301022%2C320973%2C407393%2C330401%2C239944%2C254585%2C367076%2C397560%2C359621%2C354392%2C309012%2C228698%2C258640%2C357561%2C365527%2C349145%2C327404%2C285355%2C226171%2C267937%2C363944%2C361916%2C345625%2C324853%2C292633%2C228378%2C262001%2C351966%2C345134%2C344586%2C337560%2C294427%2C228540%2C263019%2C376334%2C399055%2C369395%2C336714%2C292018%2C221217%2C252716%5D%2C%22borderColor%22%3A%22rgba%28107%2C+114%2C+128%2C+0.5%29%22%2C%22backgroundColor%22%3A%22rgba%28107%2C+114%2C+128%2C+0.05%29%22%2C%22lineTension%22%3A0.2%2C%22borderWidth%22%3A2%2C%22borderDash%22%3A%5B2%2C2%5D%2C%22radius%22%3A0%7D%2C%7B%22label%22%3A%22Sessions+in+this+year%22%2C%22data%22%3A%5B376173%2C351601%2C279658%2C312721%2C389352%2C408233%2C387882%2C391590%2C350917%2C294017%2C298776%2C413692%2C400325%2C392499%2C385895%2C361797%2C285309%2C323128%2C415132%2C432748%2C517067%2C549534%2C507360%2C389924%2C382535%2C501140%2C491676%2C468546%2C449962%2C431831%2C346292%2C394433%2C457372%2C441426%2C404434%2C395330%2C333902%2C289398%2C307281%2C407842%2C410932%2C414494%2C415727%2C395514%2C315680%2C344773%2C444457%2C441613%2C434245%2C424981%2C374942%2C323555%2C353678%2C488625%2C483740%2C473021%2C458245%2C417226%2C320100%2C343000%5D%2C%22borderColor%22%3A%22rgb%28103%2C+58%2C+183%29%22%2C%22backgroundColor%22%3A%22rgba%28103%2C+58%2C+183%2C+0.05%29%22%2C%22lineTension%22%3A0.2%2C%22borderWidth%22%3A2%2C%22radius%22%3A1%7D%5D%7D%2C%22options%22%3A%7B%22legend%22%3A%7B%22position%22%3A%22bottom%22%7D%2C%22scales%22%3A%7B%22xAxes%22%3A%5B%7B%22gridLines%22%3A%7B%22display%22%3Atrue%7D%2C%22ticks%22%3A%7B%22callback%22%3A%22%28val%29+%3D%3E+%7Breturn+val+%2B+%27k%27%3B%7D%22%7D%7D%5D%7D%7D%7D&w=700&h=428&bkg=%23ffffff&devicePixelRatio=2.0&f=png

**Comment by luctus** [comment]
hello @typpo , thanks for this project!

I copy/pasted the code [from here](https://github.com/typpo/quickchart-ruby/blob/master/examples/simple_example_with_function.rb) to my Rails console and I got this url:

https://quickchart.io/chart?c=%7B%0A++type:+%27bar%27,%0A++data:+%7B%0A++++labels:+%5B%27Q1%27,+%27Q2%27,+%27Q3%27,+%27Q4%27%5D,%0A++++datasets:+%5B%7B%0A++++++label:+%27Users%27,%0A++++++data:+%5B50,+60,+70,+180%5D%0A++++%7D,+%7B%0A++++++label:+%27Revenue%27,%0A++++++data:+%5B100,+200,+300,+400%5D%0A++++%7D%5D%0A++%7D,%0A++options:+%7B%0A++++scales:+%7B%0A++++++yAxes:+%5B%7B%0A++++++++ticks:+%7B%0A++++++++++callback:+(val)+%3D%3E+%7B%0A++++++++++++return+val+++%27k%27;%0A++++++++++%7D%0A++++++++%7D%0A++++++%7D%5D%0A++++%7D%0A++%7D%0A%7D%0A&w=600&h=300&bkg=%23ffffff&devicePixelRatio=2.0&f=png

What displays this image:
![image](https://user-images.githubusercontent.com/365101/110317560-3ad20900-8004-11eb-8d1d-f794384b2606.png)

Am I doing something wrong?

**Comment by typpo** [comment]
Hey @luctus, what version of Ruby are you using?  If this is indeed what you get using the example code, your `URI.encode_www_form` is producing different output from mine.

**Comment by typpo** [comment]
@luctus I've just published v1.1.1, which will build the URL in a hopefully more robust way: https://rubygems.org/gems/quickchart/versions/1.1.1

Can you let me know how it works?
